src/wp-includes/class-wp-query.php

```diff
@@ -3182,6 +3190,14 @@ public function get_posts() {
 					if ( 'ids' === $q['fields'] ) {
 						/** @var int[] */
 						$this->posts = array_map( 'intval', $cached_results['posts'] );
+
+						if ( $q['update_post_term_cache'] ) {
+							$prime_post_types = 'any' === $q['post_type'] ? $in_search_post_types : $q['post_type'];
+							update_object_term_cache( $this->posts, $prime_post_types );
+						}
+						if ( $q['update_post_meta_cache'] ) {
+							update_postmeta_cache( $this->posts );
+						}
```

Comment:
This logic is repeated, I wonder if it should be a method or function...

Reply:
As it's not overly complex and only used twice I figured a copy-pasta was adequate due to the use of `$in_search_post_types` which has some complex logic to determine if it's set.